holidays/countries/guatemala.py

```diff
@@ -0,0 +1,106 @@
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/dr-prodigy/python-holidays
+#  License: MIT (see LICENSE file)
+
+from datetime import date, timedelta
+from gettext import gettext as tr
+
+from holidays.calendars.gregorian import MAY, JUN, AUG, SEP, OCT, NOV
+from holidays.holiday_base import HolidayBase
+from holidays.holiday_groups import ChristianHolidays, InternationalHolidays
+
+
+class Guatemala(HolidayBase, ChristianHolidays, InternationalHolidays):
+    """
+    References:
+    - http://www.bvnsa.com.gt/bvnsa/calendario_dias_festivos.php
+    - https://www.minfin.gob.gt/images/downloads/leyes_acuerdos/decretocong19_101018.pdf
+    """
+
+    country = "GT"
+    default_language = "es"
+    supported_languages = ("en_US", "es")
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        super().__init__(*args, **kwargs)
+
+    def _add_holiday_law_19_2018(self, name: str, dt: date) -> None:
+        """
+        law 19-2018
+        https://www.minfin.gob.gt/images/downloads/leyes_acuerdos/decretocong19_101018.pdf
+        """
+        if self._is_tuesday(dt):
+            day = dt - timedelta(days=1)
+        elif self._is_wednesday(dt):
+            day = dt - timedelta(days=2)
+        elif self._is_thursday(dt):
+            day = dt + timedelta(days=4)
+        elif self._is_friday(dt):
+            day = dt + timedelta(days=3)
+        elif self._is_saturday(dt):
+            day = dt + timedelta(days=2)
+        elif self._is_sunday(dt):
+            day = dt + timedelta(days=1)
+        else:
+            day = dt
+
+        self._add_holiday(name, day)
+
+    def _populate(self, year):
+        super()._populate(year)
+
+        # New Year's Day.
+        self._add_new_years_day(tr("Año Nuevo"))
+
+        # Good Thursday.
+        self._add_holy_thursday(tr("Jueves Santo"))
+
+        # Good Friday.
+        self._add_good_friday(tr("Viernes Santo"))
+
+        # Good Saturday.
+        self._add_holy_saturday(tr("Sabado Santo"))
+
+        # Labor Day.
+        if year >= 2021:
+            self._add_holiday_law_19_2018(tr("Dia del Trabajo"), date(year, MAY, 1))
+        else:
+            self._add_labor_day(tr("Dia del Trabajo"))
+
+        # Army Day.
+        if year >= 2021:
+            self._add_holiday_law_19_2018(tr("Dia del Ejército"), date(year, JUN, 30))
+        else:
+            self._add_holiday(tr("Dia del Ejército"), JUN, 30)
+
+        # Dia de la Asunción
+        self._add_holiday(tr("Dia de la Asunción"), AUG, 15)
+
+        # Independence Day
+        self._add_holiday(tr("Día de la Independencia"), SEP, 15)
+
+        # Revolution Day
+        self._add_holiday(tr("Dia de la Revolución"), OCT, 20)
```

Comment:
Decree 19-2018 includes this holiday in movable holidays list too. Has it changed in any way?

Reply:
this is correct, but looking in the local newspapers, they say it doesn't change. or it stays at one friday. check.


https://republica.gt/vive-guatemala/feriados-2023-en-guatemala-fines-de-semana-largos-y-asuetos-de-este-ano-en-el-pais-202331414310

October 20 "Revolution Day": October 20 will fall on Friday, the law does not establish that it can be run. However, since it is a day before a weekend, Guatemalans will have a long rest of three days, since those who do not work or study on weekends will have the opportunity to rest on Friday, Saturday and Sunday.

https://www.prensalibre.com/guatemala/comunitario/asuetos-en-guatemala-2023-cuantos-asuetos-y-feriados-largos-quedan-en-lo-que-resta-del-ano/

Revolution Day: October 20 is also non-transferable, but it will be Friday, so it would have another long weekend.

Reply:
Wait, please this have a chenge from CC in EXPEDIENTE  5536-2018 i check all document. 

this is for 1 may and 20 oct.

https://leyes.infile.com/index.php?id=181&id_publicacion=81055&cmd=login

for 15 set i looking for more info.

Reply:
> this is for 1 may and 20 oct.

Yes, I see. So May 1 and Oct 20 movability is canceled from 2020, right?

Reply:
> for 15 set i looking for more info.

I said nothing about Sep 15. It's ordinary holiday, as far as I can see.

Reply:
> > this is for 1 may and 20 oct.
> 
> Yes, I see. So May 1 and Oct 20 movability is canceled from 2020, right?

yes



> > for 15 set i looking for more info.
> 
> I said nothing about Sep 15. It's ordinary holiday, as far as I can see.

I am confirming this date and looking to see if there is another decree or file, because they did not move it to Monday according to the decree.

Reply:
In summary, I see it as follows:
May 1 - move in 2019 only,
Jun 30 - move since 2019,
Oct 20 - move in 2018-2019 only.
Am I right?